Bottom: de792081e7544e180932b87bb53b263200997f15
Top:    190d2d1ac3e285e7f5af6223f202e3ddf6f490c8
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-04-05 22:30:41 +0100

Hack around Circuit CAM / EDG brokeness reading drill files










---

diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index e6bb8d5..0462fc9 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -53,10 +53,16 @@ static void gerber_fill_polygon (hidGC gc, int n_coords, int *x, int *y);
 #define gerberYOffset(pcb, y) ((long) (-(y)))
 
 /* These are for drills */
-#define gerberDrX(pcb, x) ((long) ((x)/10))
-#define gerberDrY(pcb, y) ((long) (((pcb)->MaxHeight - (y)))/10)
-#define gerberDrXOffset(pcb, x) ((long) ((x)/10))
-#define gerberDrYOffset(pcb, y) ((long) (-(y))/10)
+#ifdef EDG_ARE_BROKEN
+#  define EDG_MUL 100
+#else
+#  define EDG_MUL 1
+#endif
+
+#define gerberDrX(pcb, x) ((long) (EDG_MUL * (x)/10))
+#define gerberDrY(pcb, y) ((long) (EDG_MUL * ((pcb)->MaxHeight - (y))/10))
+#define gerberDrXOffset(pcb, x) ((long) (EDG_MUL * (x) / 10))
+#define gerberDrYOffset(pcb, y) ((long) (EDG_MUL * -(y) / 10))
 
 /*----------------------------------------------------------------------------*/
 /* Private data structures                                                    */
